### PR TITLE
Release: Update payloads for v0.10.0

### PR DESCRIPTION
**Pull request by niteeshkd**

### CODE REVIEW

**Comment by niteeshkd** [comment]
@portersrc @wainersm @fitzthum

File: config/samples/ccruntime/default/kustomization.yaml [hunk 1]
```diff
@@ -8,10 +8,9 @@ resources:
 
 images:
 - name: quay.io/confidential-containers/reqs-payload
-  newTag: latest
+  newTag: 12e8fbc72e927f9d684efe2e1214937f8b0544fc
 - name: quay.io/kata-containers/kata-deploy
-  newName: quay.io/kata-containers/kata-deploy-ci
-  newTag: kata-containers-latest
+  newTag: v3.9.0
```
Comment (stevenhorsman) [inline config/samples/ccruntime/default/kustomization.yaml:13]: This tag isn't correct - there is no `v`

File: config/samples/ccruntime/peer-pods/kustomization.yaml [hunk 2]
```diff
@@ -9,10 +9,9 @@ resources:
 
 images:
 - name: quay.io/confidential-containers/reqs-payload
-  newTag: latest
+  newTag: 12e8fbc72e927f9d684efe2e1214937f8b0544fc
 - name: quay.io/kata-containers/kata-deploy
-  newName: quay.io/kata-containers/kata-deploy-ci
-  newTag: kata-containers-latest
+  newTag: v3.9.0
```
Comment (stevenhorsman) [inline config/samples/ccruntime/peer-pods/kustomization.yaml:14]: Same here

File: config/samples/ccruntime/s390x/kustomization.yaml [hunk 3]
```diff
@@ -8,10 +8,9 @@ resources:
 
 images:
 - name: quay.io/confidential-containers/reqs-payload
-  newTag: latest
+  newTag: 12e8fbc72e927f9d684efe2e1214937f8b0544fc
 - name: quay.io/kata-containers/kata-deploy
-  newName: quay.io/kata-containers/kata-deploy-ci
-  newTag: kata-containers-latest
+  newTag: v3.9.0
```
Comment (stevenhorsman) [inline config/samples/ccruntime/s390x/kustomization.yaml:13]: and here

**Review by stevenhorsman — CHANGES_REQUESTED** [review]

File: config/samples/ccruntime/default/kustomization.yaml (hunk 1 above)
Comment (niteeshkd) [reply config/samples/ccruntime/default/kustomization.yaml]: @stevenhorsman ,  you are right. I just corrected that. Thanks!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @niteeshkd!

**Review by portersrc — APPROVED** [review]
Looks OK to me. I sanity checked it by comparing against the same step for v0.9.0 [here](https://github.com/confidential-containers/operator/pull/397/files).

File: config/samples/ccruntime/default/kustomization.yaml [hunk 4]
```diff
@@ -8,10 +8,9 @@ resources:
 
 images:
 - name: quay.io/confidential-containers/reqs-payload
-  newTag: latest
+  newTag: 12e8fbc72e927f9d684efe2e1214937f8b0544fc
 - name: quay.io/kata-containers/kata-deploy
-  newName: quay.io/kata-containers/kata-deploy-ci
-  newTag: kata-containers-latest
+  newTag: 3.9.0
```
Comment (fitzthum) [inline config/samples/ccruntime/default/kustomization.yaml:13]: I think your third commit accidentally removes the `v`.

**Review by fitzthum — COMMENTED** [review]
There still seems to be an issue with the kata version

File: config/samples/ccruntime/default/kustomization.yaml (hunk 4 above)
Comment (ldoktor) [reply config/samples/ccruntime/default/kustomization.yaml:13]: (although it should have been squashed to the previous commit. @niteeshkd could you please fix the `newTag` in the commit it's introduced in?)

**Review by ldoktor — COMMENTED** [review]
It generally looks good, there is the `v3.9.0` vs. `3.9.0` issue introduced in the 2nd commit and fixed in the 3rd one which needs to be corrected (so changes work separately).

One more thing I noticed is that there are still a couple of `3.9.0` entries here and there, shouldn't these be tackled together? `{Makefile,config/manager/kustomization.yaml,config/manifests/bases/cc-operator.clusterserviceversion.yaml,bundle/manifests/cc-operator.clusterserviceversion.yaml}`

**Comment by portersrc** [comment]
> One more thing I noticed is that there are still a couple of 3.9.0 entries here and there, shouldn't these be tackled together?
I looked at the files you pointed to, and I guess you mean the `0.9.0` entries (i.e. some lingering, hard-coded coco version numbers)? This is a good catch. I see @wainersm was [modifying these in the past](https://github.com/confidential-containers/operator/commit/ac820500a1714afa27654cb5f044b3a17c20eaa0) -- looks related to operator hub. Is now the time to fold those manual steps into this release process, or are things still too much in flux and requiring extra effort each release?

**Review by fitzthum — APPROVED** [review]
LGTM

**Comment by fitzthum** [comment]
> Is now the time to fold those manual steps into this release process, or are things still too much in flux and requiring extra effort each release?

We have treated these as a later step in the past.

**Review by mythi — APPROVED** [review]
we did some manual testing too and seems working

**Comment by wainersm** [comment]
> > One more thing I noticed is that there are still a couple of 3.9.0 entries here and there, shouldn't these be tackled together?
> 
> I looked at the files you pointed to, and I guess you mean the `0.9.0` entries (i.e. some lingering, hard-coded coco version numbers)? This is a good catch. I see @wainersm was [modifying these in the past](https://github.com/confidential-containers/operator/commit/ac820500a1714afa27654cb5f044b3a17c20eaa0) -- looks related to operator hub. Is now the time to fold those manual steps into this release process, or are things still too much in flux and requiring extra effort each release?
>

Yes, those are the changes required for publishing the released operator in operator hub. The instructions are in [here](https://github.com/confidential-containers/operator/blob/main/docs/OPERATOR_HUB.md) but I think it's a good idea to merge the steps into this doc, all in all, it's part of the release.

I don't know if step 8 could also be done simultaneously with step 5 (this). The bundle generated bundle files will point to the built operator image (`quay.io/confidential-containers/operator:v${TARGET_RELEASE}`). I don't know if the tool that build the bundle will check the existence of the image, if yes then step 8 should be done after the release.
 
For 0.10.0 release, maybe let's not change the flow.

> Edit: This is [Step 8](https://github.com/confidential-containers/confidential-containers/issues/242) for us currently.

**Comment by fitzthum** [comment]
I guess there is a nonzero chance of v0.10.1 so I will wait to tag the repo.

**Pull request merged**